datalad/plugin/wtf.py

```diff
@@ -297,35 +342,42 @@ def __call__(dataset=None, sensitive=None, clipboard=None):
         from datalad.ui import ui
         from datalad.support.external_versions import external_versions
 
-        infos = {}
+        infos = OrderedDict()
         res = get_status_dict(
             action='wtf',
             path=ds.path if ds else op.abspath(op.curdir),
             type='dataset' if ds else 'directory',
             status='ok',
             logger=lgr,
+            decor=decor,
             infos=infos,
         )
-        infos['datalad'] = _describe_datalad()
-        infos['git-annex'] = _describe_annex()
-        infos['system'] = _describe_system()
-        infos['environment'] = _describe_environment()
-        infos['configuration'] = _describe_configuration(cfg, sensitive)
-        infos['extentions'] = _describe_extensions()
-        infos['metadata_extractors'] = _describe_metadata_extractors()
-        infos['dependencies'] = _describe_dependencies()
+
+        # Define section callables which require variables.
+        # so there is no side-effect on module level original
+        section_callables = SECTION_CALLABLES.copy()
+        section_callables['location'] = partial(_describe_location, res)
+        section_callables['configuration'] = \
+            partial(_describe_configuration, cfg, sensitive)
         if ds:
-            try:
-                infos['dataset'] = _describe_dataset(ds, sensitive)
-            except InvalidGitRepositoryError as e:
-                infos['dataset'] = {"invalid": exc_str(e)}
+            section_callables['dataset'] = \
+                partial(_describe_dataset, ds, sensitive)
+        else:
+            section_callables.pop('dataset')
+        assert all(section_callables.values())  # check if none was missed
+
+        if sections is None:
+            sections = sorted(list(section_callables))
+
+        for s in sections:
+            infos[s] = section_callables[s]()
 
         if clipboard:
             external_versions.check(
                 'pyperclip', msg="It is needed to be able to use clipboard")
             import pyperclip
             report = _render_report(res)
-            pyperclip.copy(report)
+            pyperclip.copy(assure_bytes(report))
```

Comment:
Our tastes regarding collapsing unrelated changes into commits differ, so I try not to remark upon it too much, but this really should get its own commit!

Reply:
did that too ;-)